contracts/Polymorph.sol

```diff
@@ -76,25 +80,54 @@ contract Polymorph is IPolymorph, ERC721PresetMinterPauserAutoId, BMath, Reentra
         _mint(_msgSender(), tokenId);
 
         emit TokenMinted(tokenId, _genes[tokenId]);
-        emit TokenMorphed(tokenId, 0, _genes[tokenId], price, PolymorphEventType.MINT);
+        emit TokenMorphed(tokenId, 0, _genes[tokenId], polymorphPrice, PolymorphEventType.MINT);
     }
 
-    function lastTokenId() public override view returns (uint256 tokenId) {
-        return _tokenIdTracker.current();
+    function bulkBuy(uint256 amount) public override payable nonReentrant {
+        require(msg.value >= polymorphPrice.mul(amount), "Not enough funds");
+        require(amount <= 30, "Cannot bulk buy more than 30");
```

Comment:
Lets make the 30 a parameter like the premint and make it changeable by the DAO

Reply:
Changed as well